repo/repo.go

```diff
@@ -25,6 +25,8 @@ type Repo interface {
 	// SetAPIAddr sets the API address in the repo.
 	SetAPIAddr(addr string) error
 
+	Self() Repo
```

Comment:
`node.Repo` is _not_ a `FSRepo` it is a `repo.ref`.  `repo.ref` is defined here: https://github.com/ipfs/go-ipfs/blob/c067fb9e83e89cf04226d2c43de7c6fd5ebbccd2/repo/onlyone.go#L50.  It may be easier for you to just try it without the Self() and see for yourself.